File: src/main/java/com/zl/pleasetweakwindows/Tweak.java
package com.zl.pleasetweakwindows;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Tweak {
    private final String title;
    private final String applyScript;
    private final String revertScript;
    private final List<SubTweak> subTweaks;

    public Tweak(String title, String applyScript, String revertScript) {
        this.title = title;
        this.applyScript = applyScript;
        this.revertScript = revertScript;
        this.subTweaks = new ArrayList<>();
    }

    public void addSubTweak(SubTweak subTweak) {
        subTweaks.add(subTweak);
    }

    public String getTitle() {
        return title;
    }

    public String getApplyScript() {
        return applyScript;
    }

    public String getRevertScript() {
        return revertScript;
    }

    public List<SubTweak> getSubTweaks() {
        return Collections.unmodifiableList(subTweaks);
    }
}

File: src/PleaseTweakWindows/Models/Tweak.cs
using System.Collections.Immutable;

namespace PleaseTweakWindows.Models;

public sealed record Tweak(
    string Title,
    string ApplyScript,
    string RevertScript,
    ImmutableList<SubTweak> SubTweaks)
{
    public IEnumerable<SubTweak> RunAllSubTweaks =>
        SubTweaks.Where(sub =>
            sub.Type == SubTweakType.Toggle &&
            !string.IsNullOrWhiteSpace(sub.RevertAction));

    public bool CanRunAll => RunAllSubTweaks.Any();

    /// <summary>Convenience ctor for categories whose Apply and Revert run the same script.</summary>
    public Tweak(string title, string script, ImmutableList<SubTweak> subTweaks)
        : this(title, script, script, subTweaks) { }
}
